docs/testing-styleguide.MD

```diff
@@ -0,0 +1,278 @@
+
+### Functional Browser Testing
+#### Manual Testing
+Every new or amended component should be fully tested in either Storybook and/or [codesandbox.io](https://codesandbox.io) using functional testing, UI testing and internationalization testing techniques. We use [Jira](https://www.atlassian.com/software/jira) for issue and project tracking. 
+
+##### Browser Support
+Carbon is supported in the following browsers:
+* Chrome
+* Firefox
+* Safari
+* Edge Chromium
+
+Components should be tested in all supported browsers unless the piece of functionality is not specifically related to a UI feature, e.g. decimal precision.
+
+##### Jira
+For every testable task, a test session needs to be created for each of the supported browsers. Results should be added to each of the conditions in the created test sessions with a pass `(/)` or fail `(x)` mark.
+
+Any issues found during testing should be raised as a "Defect" in Jira unless the ticket is already of a “Bug” type, in which case comments and screenshots/recordings/GIFs can be added to the existing ticket.
+
+If the issue is present on the master branch you should first check if the issue has already been logged in GitHub. If it is present, it should be upvoted with a “+1” comment. If it has not been logged, details of how to do this are contained in the [CONTRIBUTING.MD](https://github.com/Sage/carbon/blob/master/CONTRIBUTING.md) guide. Collaborators internal to Sage should also log a Jira ticket for the issue as a "Bug" type with the GitHub issue linked. However, there should be no link from the GitHub issue to the Jira ticket.
+
+##### Storybook
+Storybook forms our Carbon component library documentation. The component stories are written in `MDX` format containing as many examples as required to showcase our components and with minimal knobs for changing prop states. Every component should have at least one story (named `default`/`multiple`/`validation` etc.) located in the `Design System` directory with `Docs`, `Notes` and `Accessibility` sections. Where required, it should be possible to test all events, e.g. `onClick`, `drag` etc, in these stories. Results of events tests can be found in the `Actions` tab.
+
+Where the stories in the main `Design System` documentation directory do not cover all scenarios required to fully test the component, additional stories must be created in the `Test` directory. For example, scenarios testing special characters, all combinations of size/type/state etc should have stories in the `Test` directory under the relevant component and named `visual`. A developer may need to assist with writing the additional required stories here.
+
+To clarify the purpose of each directory,
+
+| Directory     | Usage                                                                    |
+|---------------|--------------------------------------------------------------------------|
+| Design System | User facing documentation, primary visual regression tests               |
```

Comment:
We need to state that Cypress regression (non-visual) should be exercising the Design System stories where possible. `Test` should just be if we need something that the documentation doesn't show (this shouldn't be often).

Reply:
I'll amend this line (91) in the table to read "User facing documentation, Cypress regression and visual tests". Line 85 just above explains that "Where the stories in the main `Design System` documentation directory do not cover all scenarios required to fully test the component, additional stories must be created in the `Test` directory."